src/plots/smith/smith.js

```diff
@@ -0,0 +1,1456 @@
+    // offsets from paper edge to layout domain box
+    _this.xOffset = gs.l + gs.w * xDomain[0];
+    _this.yOffset = gs.t + gs.h * (1 - yDomain[1]);
+    // lengths of the layout domain box
+    var xLength = _this.xLength = gs.w * (xDomain[1] - xDomain[0]);
+    var yLength = _this.yLength = gs.h * (yDomain[1] - yDomain[0]);
+    // sector to plot
+    var sector = smithLayout.sector;
+    _this.sectorInRad = sector.map(deg2rad);
+    var sectorBBox = _this.sectorBBox = computeSectorBBox(sector);
+    var dxSectorBBox = sectorBBox[2] - sectorBBox[0];
+    var dySectorBBox = sectorBBox[3] - sectorBBox[1];
+    // aspect ratios
+    var arDomain = yLength / xLength;
+    var arSector = Math.abs(dySectorBBox / dxSectorBBox);
+    // actual lengths and domains of subplot box
+    var xLength2, yLength2;
+    var xDomain2, yDomain2;
+    var gap;
+    if(arDomain > arSector) {
+        xLength2 = xLength;
+        yLength2 = xLength * arSector;
+        gap = (yLength - yLength2) / gs.h / 2;
+        xDomain2 = [xDomain[0], xDomain[1]];
+        yDomain2 = [yDomain[0] + gap, yDomain[1] - gap];
+    } else {
+        xLength2 = yLength / arSector;
+        yLength2 = yLength;
+        gap = (xLength - xLength2) / gs.w / 2;
+        xDomain2 = [xDomain[0] + gap, xDomain[1] - gap];
+        yDomain2 = [yDomain[0], yDomain[1]];
+    }
+    _this.xLength2 = xLength2;
+    _this.yLength2 = yLength2;
+    _this.xDomain2 = xDomain2;
+    _this.yDomain2 = yDomain2;
+    // actual offsets from paper edge to the subplot box top-left corner
+    var xOffset2 = _this.xOffset2 = gs.l + gs.w * xDomain2[0];
+    var yOffset2 = _this.yOffset2 = gs.t + gs.h * (1 - yDomain2[1]);
+    // circle radius in px
+    var radius = _this.radius = xLength2 / dxSectorBBox;
+    // circle center position in px
+    var cx = _this.cx = xOffset2 - radius * sectorBBox[0];
+    var cy = _this.cy = yOffset2 + radius * sectorBBox[3];
+    // circle center in the coordinate system of plot area
+    var cxx = _this.cxx = cx - xOffset2;
+    var cyy = _this.cyy = cy - yOffset2;
+
+    _this.radialAxis = _this.mockAxis(fullLayout, smithLayout, radialLayout, {
+        _id: 'realaxis2',
+        side: {
+            counterclockwise: 'top',
+            clockwise: 'bottom'
+        }[radialLayout.side],
+        // keep track of real side
+        _realSide: radialLayout.side,
+        domain: [0, 100]
+    });
+
+    // _this.radialAxis = _this.mockAxis(fullLayout, smithLayout, radialLayout, {
+        // // make this an 'x' axis to make positioning (especially rotation) easier
+        // _id: 'x',
+        // // convert to 'x' axis equivalent
+        // side: {
+            // counterclockwise: 'top',
+            // clockwise: 'bottom'
+        // }[radialLayout.side],
+        // // keep track of real side
+        // _realSide: radialLayout.side,
+        // // spans length 1 radius
+        // domain: [innerRadius / gs.w, radius / gs.w]
+    // });
```

Comment:
This commented block could be removed.

Reply:
8d9767111cc9510dfb50cc4bff3a84bb0f9af40f